scio-parquet/src/main/scala/com/spotify/scio/parquet/avro/ParquetAvroIO.scala

```diff
@@ -230,10 +233,9 @@ object ParquetAvroIO {
       // org.apache.beam.sdk.coders.AvroCoder.
       if (ScioUtil.classOf[A] == classOf[GenericRecord]) {
         job.getConfiguration.setBoolean("parquet.avro.compatible", false)
-        job.getConfiguration.set(
-          "parquet.avro.data.supplier",
-          "org.apache.parquet.avro.GenericDataSupplier"
-        )
+        AvroReadSupport.setAvroDataSupplier(job.getConfiguration, classOf[GenericDataSupplier])
+      } else {
+        AvroReadSupport.setAvroDataSupplier(job.getConfiguration, classOf[LogicalTypeSupplier])
```

Comment:
I'm a bit torn on setting this as the default for all SR parquet-avro operations, regardless if the schema contains a logical type or not; but it should be harmless, just a small in-memory map that never gets used.

The downside of making this an opt-in feature is that the default behavior of `.saveAsAvroFile` w.r.t LogicalTypes will remain different from the default behavior of `.saveAsParquetAvroFile`; users wouldn't be able to migrate from Avro to Parquet-Avro with just the drop-in replacement. 🤔

Maybe we could scan the `schema` param for any logical-type fields, and if we detect any, only *then* do we set the data-supplier to `LogicalTypeSupplier`. If we don't find any, we let it default to just `SpecificDataSupplier`.

Reply:
maybe we can start with this simpler solution, and optimize it when it's well tested?

Reply:
probably the simplest solution is making this totally opt in for any users whose schema contains a logical type :) I'm just trying to reason if there's any downside to always including logical-type conversions